### PR TITLE
[risk=no] [RW-7427] Create Search Table From VAT

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
1. Added JSON Schema for the search table.
2. Added commands to generate table from pipeline when flag is true.

### CODE REVIEW

File: public-api/db-cdr/generate-cdr/make-bq-data.sh [hunk 1]
```diff
@@ -100,6 +108,109 @@ do
...
 
+if [ "$SEARCH_VAT" = true ]; then
+  bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+  "DROP MATERIALIZED VIEW IF EXISTS \`$OUTPUT_PROJECT.$GENOMICS_DATASET.wgs_variant_vid_cluster_mv\`"
+
+  bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+  "DROP TABLE IF EXISTS \`$OUTPUT_PROJECT.$GENOMICS_DATASET.wgs_variant\`"
+
+  bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+  "CREATE TABLE \`$OUTPUT_PROJECT.$GENOMICS_DATASET.wgs_variant\` as
```
Comment (freemabd) [inline public-api/db-cdr/generate-cdr/make-bq-data.sh]: This should be a clustered table correct? In order for the materialized view clustering to work, the base table has to be clustered as well.

File: public-api/db-cdr/generate-cdr/make-bq-data.sh [hunk 2]
```diff
@@ -100,6 +108,109 @@ do
...
+     sorted_transcripts.ref_allele,
+     sorted_transcripts.alt_allele,
+     sorted_transcripts.cons_str,
+     genes.genes
+  FROM sorted_transcripts, genes
+  WHERE genes.vid = sorted_transcripts.variant_id
+  AND (sorted_transcripts.row_number =1 or sorted_transcripts.transcript is NULL);"
+
+  bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+  "DROP TABLE IF EXISTS \`$OUTPUT_PROJECT.$GENOMICS_DATASET.wgs_variant_vid_cluster\`"
```
Comment (freemabd) [inline public-api/db-cdr/generate-cdr/make-bq-data.sh]: you don't need this.

File: public-api/db-cdr/generate-cdr/make-bq-data.sh [hunk 3]
```diff
@@ -100,6 +108,109 @@ do
...
+
+  bq mk --table --schema=$schema_path/wgs_variant_vid_cluster.json --time_partitioning_type DAY --clustering_fields variant_id aou-db-prod:genomics_v1.wgs_variant_vid_cluster
```
Comment (freemabd) [inline public-api/db-cdr/generate-cdr/make-bq-data.sh]: This isn't needed. This should be a MV created from the base clustered table.

File: public-api/db-cdr/generate-cdr/make-bq-data.sh [hunk 4]
```diff
@@ -100,6 +108,109 @@ do
...
+
+  bq --quiet --project=$BQ_PROJECT query --nouse_legacy_sql \
+  "insert into \`$OUTPUT_PROJECT.$GENOMICS_DATASET.wgs_variant_vid_cluster\`
```
Comment (freemabd) [inline public-api/db-cdr/generate-cdr/make-bq-data.sh]: This should be a reclustered materialized view off the base table wgs_variant. You do this in the next step. You only need the MV but not this concrete table.

File: public-api/libproject/devstart.rb [hunk 5]
```diff
@@ -583,6 +583,11 @@ def generate_public_cdr_counts(cmd_name, *args)
       ->(opts, v) { opts.bucket = v},
       "GCS bucket required."
     )
+    op.add_option(
+      "--search-vat [search-vat]",
+      ->(opts, v) { opts.search_vat = v},
+      "Flag to generate search table from VAT. Optional."
+    )
```
Comment (freemabd) [inline public-api/libproject/devstart.rb:590]: You should add a default value here. This will allow you to remove line 49 from make-bq-data.sh

**Review by freemabd — COMMENTED** [review]
Added a few comments about your tables. In order to take advantage of projection materialized views you need an underlying clustered table(wgs_variant). Then you can create any materialized view from that base view as... 
`CREATE MATERIALIZED VIEW dataset.mv
CLUSTER BY newClusteredId
AS SELECT * FROM dataset.base_table;`

File: public-api/db-cdr/generate-cdr/make-bq-data.sh (hunk 1 above)
Comment (SrushtiGangireddy) [reply public-api/db-cdr/generate-cdr/make-bq-data.sh]: @freemabd Used this wgs_variant_vid_cluster as base for MV. I guess we just need one table changing this.

File: public-api/db-cdr/generate-cdr/make-bq-data.sh (hunk 2 above)
Comment (SrushtiGangireddy) [reply public-api/db-cdr/generate-cdr/make-bq-data.sh]: Deleted this. Thanks.

File: public-api/db-cdr/generate-cdr/make-bq-data.sh (hunk 3 above)
Comment (SrushtiGangireddy) [reply public-api/db-cdr/generate-cdr/make-bq-data.sh]: Deleted this. Thanks

File: public-api/db-cdr/generate-cdr/make-bq-data.sh (hunk 4 above)
Comment (SrushtiGangireddy) [reply public-api/db-cdr/generate-cdr/make-bq-data.sh]: Changed this.

File: public-api/libproject/devstart.rb (hunk 5 above)
Comment (SrushtiGangireddy) [reply public-api/libproject/devstart.rb:590]: Added default value in here.

**Review by freemabd — APPROVED** [review]
Looks good.. thanks

**Pull request merged**